File: python/linear_search.py
"""
    Here we are going to implement the linear search algorithm
"""


def linear_search(lst, target):
    for i in range(0, len(lst)):
        if lst[i] == target:
            return i
    return -1


def verify_linear(index):
    if (index != -1):
        print(f"The target is found at index {index}")
    else:
        print("The target does not exist in the list")

File: javascript/linear_search.js
`
    Implementation of the linear search algorithm in JS
        Time complexity : O(n)
        Space complexity: O(1)
`

const arr = [1, 3, 4, 5, 6, 7, 8, 9, 12, 14, 16, 18, 20]
const t = 200

const linear_search = (array, target)=> {
    for(let i = 0; i < array.length; i++) {
        if(array[i] === target) return i
    }
    return -1
}

const verify = (index) => {
    if (index === -1) {
        console.log("The target does not exist in the list")
        return
    }
    console.log(`The target is found at the index ${index}`);
}



const start = performance.now()
const index = linear_search(arr, t)
const end = performance.now()

verify(index)
console.log(`RUNTIME: ${end-start}`)